### ISSUE TITLE
[FEATURE REQUEST] Different treatment for packages with denied upgrades

### ISSUE THREAD

**Issue by aaronliu0130** [marticliment/WingetUI#1812]
### Please confirm these before moving forward

- [X] I have searched for my feature proposal and not found a work-in-progress/duplicate/resolved/discarded issue.
- [X] This proposal is a completely new feature. If you want to suggest an improvement or an enhancement, please use [this template](https://github.com/marticliment/WingetUI/issues/new?assignees=marticliment&labels=enhancement&projects=&template=enhancement-improvement.yml&title=%5BENHANCEMENT%5D+%28Enter+your+description+here%29)

### Describe the new feature

In [scheme 1.6](https://github.com/microsoft/winget-pkgs/blob/master/doc/manifest/schema/1.6.0/installer.md), the "deny" option was added to UpgradeBehavior to specify that WinGet shouldn't attempt to upgrade the package. It'd be nice if such packages weren't attempted by WingetUI to upgrade every day (such as an auto-skip along with it being pinned, perhaps?) along with a "skipped" icon being displayed for it.

### Describe how this new feature could help users.

It would be helpful to automatically

**Comment by marticliment** [comment]
I don't really undertstand the point of it. Could you please give an example?

**Comment by aaronliu0130** [comment]
Sorry, I meant schema 1.6.
For example, if I have an old version of https://github.com/PckgrBot/winget-pkgs/blob/master/manifests/d/Discord/Discord/1.0.9034/Discord.Discord.installer.yaml installed and do `winget upgrade Discord.Discord`, winget will say
```
The package cannot be upgraded using winget. Please use the method provided by the publisher for upgrading this package.
```
WingetUI reports that as an error during updating, and replaces the package's icon with a yellow circle that has something in it. I think Winget should automatically skip that version of the package, replace the package's icon with the skip icon, send an appropriate notification, and persist the bottom status bar for that package (even across reboots) until the package is at the latest version or the user dismisses it.

**Comment by marticliment** [comment]
I see what you mean now. In my opinion, what I'd do is showing a custom error message, in the same way in which when a hash check fails or when the system requires a reboot the user gets warned about it, a custom error message explainig the issue could be shown, explaining to the user the situation and what must be done.

**Comment by Kuyumee** [comment]
Would be nice if WingetUI will "Skip this version" when it receives **ANY** kind of error during upgrades

**Comment by aaronliu0130** [comment]
I disagree. Errors usually need user intervention, and just ignoring it could make everything look up-to-date already

**Comment by Kuyumee** [comment]
What I mean is- WingetUI should not retry to upgrade it again without user intervention. Not "Skip this version", but something similar.

**Comment by aaronliu0130** [comment]
It might be environmental factors, such as having the application open, that can be eradicated on retry

**Comment by marticliment** [comment]
I agree with @aaronliu0130 about this.
